api/rpc/stack/parse.go

```diff
@@ -9,23 +9,13 @@ import (
 type serviceMap struct {
 	Image       string      `yaml:"image"`
 	Ports       []string    `yaml:"ports"`
-	Replicas    int         `yaml:"replicas"`
+	Replicas    int32       `yaml:"replicas"`
 	Environment interface{} `yaml:"environment"`
 	Labels      interface{} `yaml:"labels"`
 }
 
-// Service represents a docker service for use in a docker stack
-type Service struct {
-	Name        string
-	Image       string
-	Ports       map[nat.Port][]nat.PortBinding
-	Replicas    int
-	Environment map[string]string
-	Labels      map[string]string
-}
-
-func parseStackYaml(in string) (out []Service, err error) {
-	out = []Service{}
+func parseStackYaml(in string) (out Stack, err error) {
+	out = Stack{}
```

Comment:
Minor thing, but I'd save the value copying for the return and do this instead;  `out = &Stack{}`